### DISCUSSION TITLE
我有些疑问关于umi4使用构建缓存

### DISCUSSION THREAD

**Discussion by dddssw** [umijs/umi#12244] [Q&A]
![image](https://github.com/umijs/umi/assets/58782768/39c54008-e1c9-4b1b-8e08-1ae25370d382)
如果webpack5开启cache，我修改业务代码，与之相关的缓存应该失效了。再次打包产物应该是新的吧。为什么产物可能是旧的呢，我感到疑惑

**Comment by fz6m** [comment]
会干扰项目构建结果的因素有很多，比如所有关联的配置文件（如 umi 配置文件等）、package.json 、tsconfig.json 、一些企业内部特定的文件、环境变量等等，一旦有一个发生改变，都应该让缓存失效，否则构建出来的产物就是旧的，这是非常严重的。

但如何判断该何时失效，关联的文件有哪些，每个人的项目里有哪些配置文件，这些在框架层面是不知道的，所以没法默认开启这个物理缓存的功能，所以如果你们需要物理缓存，应该阅读了解相关说明文档之后，确保没有风险后再开启。

**Reply by dddssw** [reply]
如果排除配置文件，环境变量等，只是修改页面上的业务代码webpack应该会知道对应的缓存会过期下次打包重新构建这部分吧

**Reply by fz6m** [reply]
如果是源码改动是没问题的，只需要 CICD 里能自动下载上次的物理缓存即可。
